### ISSUE TITLE
Cross-Site Scripting Vectors (XSS)

### ISSUE THREAD

**Issue by cferdinandi** [justin-schroeder/arrow-js#78]
Hi there! From [one lightweight reactive library creator](https://reefjs.com) to another, nice work!

Just wanted to flag a few XSS vectors not currently blocked by ArrowJS. The first two alerts show up. The third alert (in the svg) and fourth do _not_. The janky table format is rendered, as is the link which, if opened, runs an attack.

I'm not sure what your rendering and diffing engine looks like under-the-hood, or I'd share a fix, but if you want, feel free to peruse my source code if anything in there is useful to you.

```js
let xss = [
	`<p><img src="x" onerror="alert(1)"></p>`,
	`<p>abc<iframe//src=jAva&Tab;script:alert(2)>def</p>`,
	`<svg><g/onload=alert(3)//<p>`,
	`<math><mi//xlink:href="data:x,<script>alert(4)</script>">`,
	`<TABLE><tr><td>HELLO</tr></TABL>`,
	`<a href="data:text/html;base64,PHNjcmlwdD5hbGVydCgiSGVsbG8iKTs8L3NjcmlwdD4=">test</a>`
];

let template = html`${xss.map(function (elem) {
	return html`${elem}`;
})}`;

template(document.querySelector('#app'));
```

**Comment by madelson** [comment]
Will these be fixed?

**Comment by justin-schroeder** [comment]
Definitely. It actually is fixed in the `arrow-refactor-2` branch (as well as a bunch of perf improvements). However, that said, arrow is still an experiment and I’m looking for the right tradeoffs to make. There will likely be a `arrow-refactor-3` branch perhaps even a 4th as we try to path find here.

**Comment by cferdinandi** [comment]
This is also generally only a problem with third party data. Most people don’t XSS attack themselves. Still useful to be aware of.

**Comment by madelson** [comment]
@justin-schroeder thanks for the update. Do you have a sense of when the next version would be released / how breaking it would be? Is there a good way to follow progress? I'm starting a new project and interested in trying Arrow. Curious what the upcoming changes and timelines are like.

@cferdinandi the code you show has these hard-coded into the templates themselves. I'm more worried about an XSS attack where I include some data as text content and arrow fails to escape it such that it injects tags/behavior onto the page. Have you seen anything like that? Am I thinking about this the right way?

**Comment by cferdinandi** [comment]
@madelson That's exactly right! I hard-coded them just for example purposes, but in real-life situations, the danger is user-generated or API-derived data that you use in your templates containing malicious code.

**Comment by pwFoo** [comment]
I like both Reef and arrow-js, but maybe prefer the `$on` way of arrow-js.
Is that fixed in latest version?

**Comment by graue** [comment]
No, Arrow is still vulnerable to this attack.

**Comment by justin-schroeder** [comment]
This is solved in 1.0 👍

https://arrow-js.com/play/?id=8171f54793e9b5de6163821d794a1cc7
